code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -262,22 +376,60 @@
 	var/datum/limb/L = get_limb(target_zone)
 
 	if (!L || (L.limb_status & LIMB_DESTROYED))
-		return
+		to_chat(X, "<span class='xenodanger'>We can't punch that which doesn't exist!</span>")
+		return FALSE
+
+	var/stagger_stacks = 3
+	var/slowdown_stacks = 3
+	var/punch_description = "powerful"
+	var/S = pick('sound/weapons/punch1.ogg','sound/weapons/punch2.ogg','sound/weapons/punch3.ogg','sound/weapons/punch4.ogg')
 
-	X.visible_message("<span class='xenowarning'>\The [X] hits [src] in the [L.display_name] with a devastatingly powerful punch!</span>", \
-		"<span class='xenowarning'>We hit [src] in the [L.display_name] with a devastatingly powerful punch!</span>", visible_message_flags = COMBAT_MESSAGE)
+	if(pulledby == X) //If we're being grappled by the Warrior punching us, it's gonna do extra damage and debuffs; combolicious
+		damage *= 2
+		slowdown_stacks *= 2
+		stagger_stacks *= 2
+		ParalyzeNoChain(0.5 SECONDS)
+		punch_description = "devastating"
+
+	playsound(src, S, 50, 1)
+	X.face_atom(src) //Face the target so you don't look like an idiot
+	X.do_attack_animation(src, ATTACK_EFFECT_YELLOWPUNCH)
```

Comment:
Src is a datum

Reply:
??? src in this case is a human, though I want to expand this to include other types